flow/packet-headers/macsec.yaml

```
@@ -0,0 +1,14 @@
+components:
+  schemas:
+    Flow.Macsec:
+      description: MACsec packet header.
+      type: object
+      properties:
+        choice:
+          description: Currently only auto choice is allowed. If choice is auto, MACsec header is autogenerated. In future, custom choice will be given to set header fields.
```

1) I think we can mention that if auto choice is selected, flow.tx_rx.choice must be of type 'device' and flow.tx_rx.device.tx_names[0] must be chosen to be an endpoint that is on or behind a MACSec enabled ethernet to be able to correctly auto-fill the fields of the MACsec header . If either of this is not true, the implementation should return an error specifying the issue . 
2) Maybe : "a custom choice can be added to allow user to control specific MACsec header fields or/and to generate flow.tx_rx.port type of traffic with MACSec header with all fields explicitly specified by the user."

---

I have updated description with above information.